scripts/versioner.py

```diff
@@ -172,8 +175,15 @@ def git_ancestor(maybe_ancestor, descendant, repo_dir=MOOSE_DIR):
     @staticmethod
     def git_file(file, commit, repo_dir=MOOSE_DIR, allow_missing=False):
         """ gets the contents of a file at a given git commit """
-        repo_dir = repo_dir.rstrip(os.sep)
-        file = file.replace(repo_dir, '.')
+        # If a relative path is supplied, prefix repo_dir
+        if not os.path.exists(file):
+            if os.path.exists(os.path.join(repo_dir, file)):
+                file = os.path.join(repo_dir, file)
```

Comment:
```python
        # If a relative path is supplied, prefix repo_dir
        if not os.path.exists(file):
            if os.path.exists(os.path.join(repo_dir, file)):
                file = os.path.join(repo_dir, file)
```
My gut tells me this is not 100% safe. We are assuming the incoming relative path'd `file` was meant to exist in `repo_dir`. What if it wasn't, and we are now working with a the user did not intend...